src/lib/seam/connect/seam-http-request.ts

```diff
@@ -0,0 +1,115 @@
+import type { Method } from 'axios'
+
+import type { Client } from './client.js'
+import { SeamHttpActionAttempts } from './index.js'
+import type { SeamHttpRequestOptions } from './options.js'
+import { resolveActionAttempt } from './resolve-action-attempt.js'
+
+export interface SeamHttpRequestParent {
+  readonly client: Client
+  readonly defaults: Required<SeamHttpRequestOptions>
+}
+
+export interface SeamHttpRequestConfig<TBody> {
+  url?: string
+  method?: Method
+  params?: any
+  data?: TBody
```

Comment:
For naming ref the diagram here: https://nodejs.org/api/url.html#url-strings-and-url-objects

I realized supporting URL might be tricky since it depends on Axios interceptors and the config. Same with params. If we eventually switch to GET, body will be undefined and all the params would be contained in the URL `query` instead. Really, we want Axios to support a thenable as well and build this module on top of that.

Since we don't use GET yet, none of the URLs would have a `query`, so  `pathname` and `path` will always be equivalent, thus we can support

```suggestion
  path: string
  method: Method
  body?: TBody
```

Reply:
do we want to diverge from what axios does here given that axios is already part of the api?

Reply:
Axios is only part of the API via `client` and interceptors and prefixed options, all are advanced use cases. IMO this is `SeamHttpRequest` not `AxiosRequestConfig` and we should leverage the URL standard here, not Axios which is non-standard.

Reply:
we actually do use `get` in one place: `connectWebviews.view()`

Reply:
I think --^ does complicate this a little, do we add `search` and `searchParams`?